# Question 14
-----

In [1]:
from scipy.integrate import quad
import math

class normal:
    def __init__(self, u, sigma):
        self.u = u
        self.sigma = sigma
        self.root2pi = math.sqrt((2 * math.pi)) 
    
    def normal_func(self, x):
        return (1 / (self.sigma * self.root2pi)) * math.exp((-1 * pow((x - self.u), 2)) / (2 * pow(self.sigma, 2)))

    def integrate(self, lower_limit, upper_limit):
        result, error = quad(self.normal_func, lower_limit, upper_limit)
        
        return result

    def locate(self, area):
        fx = 0
        x = -6
        while fx < area:
            fx = self.integrate(-6, x)
            x = x + 0.00001
            
        return x

In [2]:
z = normal(0, 1)
loc_1 = z.locate(0.04)
loc_2 = z.locate(0.97)
print(loc_1)
print(loc_2)

-1.7506700000609796
1.880809999945876


$$1.75\sigma + 53 = \mu$$

$$65 - 1.88\sigma = \mu$$

$$1.75\sigma + 53 = 65 - 1.88\sigma$$

$$3.63\sigma = 12$$

$$\sigma = 3.31$$

$$\mu = (1.75 * 3.31) + 53$$

$$\mu = 58.8$$

In [3]:
dist = normal(58.8, 3.31)
x = 0
fx = 0
area = 0.25
while fx < area:
    fx = dist.integrate(0, x)
    x = x + 0.01
lower = x

x = 0
fx = 0
area = 0.75
while fx < area:
    fx = dist.integrate(0, x)
    x = x + 0.01
upper = x

intqrange = upper - lower
print("interquartile range = " + str(intqrange))

interquartile range = 4.469999999999111
